src/uart.js

```diff
@@ -98,6 +109,53 @@ function UART(cpu, port, bus)
         this.data_received(data);
     }, this);
 
+    this.bus.register("serial" + this.com + "-modem-status-input", function(data)
+    {
+        this.set_modem_status(data);
+    }, this);
+
+    // Set individual modem status bits
+
+    this.bus.register("serial" + this.com + "-carrier-detect-input", function(data)
+    {
+        if(data) {
+            this.set_modem_status(this.modem_status | (1 << UART_MSR_DCD) | (1 << UART_MSR_DDCD));
+        }
+        else {
+            this.set_modem_status(this.modem_status & ~(1 << UART_MSR_DCD) & ~(1 << UART_MSR_DDCD));
+        }
+    }, this);
+
+    this.bus.register("serial" + this.com + "-ring-indicator-input", function(data)
+    {
+        if(data) {
+            this.set_modem_status(this.modem_status | (1 << UART_MSR_RI) | (1 << UART_MSR_TERI));
+        }
+        else {
+            this.set_modem_status(this.modem_status & ~(1 << UART_MSR_RI) & ~(1 << UART_MSR_TERI));
+        }
+    }, this);
+
+    this.bus.register("serial" + this.com + "-data-set-ready-input", function(data)
+    {
+        if(data) {
+            this.set_modem_status(this.modem_status | (1 << UART_MSR_DSR) | (1 << UART_MSR_DDSR));
+        }
+        else {
+            this.set_modem_status(this.modem_status & ~(1 << UART_MSR_DSR) & ~(1 << UART_MSR_DDSR));
+        }
+    }, this);
+
+    this.bus.register("serial" + this.com + "-clear-to-send-input", function(data)
+    {
+        if(data) {
+            this.set_modem_status(this.modem_status | (1 << UART_MSR_CTS) | (1 << UART_MSR_DCTS));
+        }
+        else {
+            this.set_modem_status(this.modem_status & ~(1 << UART_MSR_CTS) & ~(1 << UART_MSR_DCTS));
+        }
```

Comment:
I'd say these are better written like this:

```js
const status = data ?
    this.modem_status | (1 << UART_MSR_CTS) | (1 << UART_MSR_DCTS) : 
    this.modem_status & ~(1 << UART_MSR_CTS) & ~(1 << UART_MSR_DCTS);
this.set_modem_status(status);
```

Reply:
Changed, thanks!